src/ZkFold/Base/Algebra/Basic/Class.hs

```diff
@@ -137,8 +139,24 @@ rootOfUnity l
                 x' = x ^ ((order @a - 1) `div` n)
             in bool (rootOfUnity' g') x' (x' ^ (n `div` 2) /= one)
 
+--}
 --------------------------------------------------------------------------------
 
+instance AdditiveSemigroup Int where
```

Comment:
Why do we need these instances? (Just wanted to note that they are not lawful as `Int` overflows)

Reply:
This was used for testing. I was tired of writing qualified Prelude imports each time I needed to do arithmetic on Ints. I'll remove it now.